feat(supernets): standards and cross-chain transactions based on v0.9 (#1627)
maticnetwork/matic-docs: docs/supernets/design/assets/erc/erc20.md

## ChildERC20

[`ChildERC20`](/docs/supernets/interfaces/erc20/childerc20.md) is a token standard that enables developers to create fungible tokens on both the Supernet and the associated rootchain. Developers can use the `ChildERC20` contract as a template to create a new `ChildERC20` token and deploy it on their Supernet. The contract requires a name, symbol, and number of decimals to determine the minimum unit of the token.

`ChildERC20` tokens are minted and burned on the Supernet through the corresponding [`ERC20Predicate`](/docs/supernets/interfaces/erc20/childerc20-predicate.md) contract. This ensures that the supply of tokens on the rootchain and the Supernet remains in sync. The `ERC20Predicate` contract also allows for the transfer of tokens between the two networks using the native bridge.

## NativeERC20

> [`NativeERC20`](/docs/supernets/interfaces/erc20/native-erc20.md) is the native token contract used on Supernets. It is designed for the Supernet and offers fast and inexpensive transactions. `NativeERC20` is deployed on the network's rootchain and is managed through the System contract. It can be minted and burned by the Predicate contract.

Reviewer: ```suggestion
[`NativeERC20`](/docs/supernets/interfaces/erc20/native-erc20.md) is the native token contract used on Supernets. It is designed for the Supernet and offers fast and inexpensive transactions. `NativeERC20` is deployed on the Supernets and it relies on native transfer precompile to make transfers themselves. It can be minted and burned by the Predicate contract.
```

> In addition, `NativeERC20Mintable` is the contract used to create and manage `NativeMintable` tokens, which are non-fungible tokens that represent assets on the Supernet. These tokens can be managed through the native bridge contract and transferred between a Supernet and rootchain networks.

Reviewer: ```suggestion
In addition, `NativeERC20Mintable` is the contract used to create and manage `NativeMintable` tokens, which are fungible tokens that represent assets on the Supernet. These tokens can be managed through the native bridge contract and transferred between a Supernet and rootchain networks.
```

### EIP1559Burn

> The `EIP1559Burn` contract is an additional feature on the Polygon Supernets that allows users to convert their `NativeERC20` tokens into ERC-20 tokens. This contract is useful for users who want to move their native tokens to the Polygon PoS network or other networks that support ERC-20 tokens. To use the `EIP1559Burn` contract, users first need to initialize the contract by specifying the address of the `ChildERC20Predicate` contract on the Supernet and the destination address on the rootchain where the ERC-20 tokens will be sent after burning. Once initialized, users can call the `withdraw` function, which will burn the native tokens and send the equivalent amount of ERC-20 tokens to the specified destination address.

Reviewer: It is true that we have it on `core-contracts` repo, but it isn't integrated into the Edge. Should we remove it from the docs?

Author: Updated.

## Differences between ChildERC20 and NativeERC20

> The main difference between `ChildERC20` and `NativeERC20` is their deployment and management. `ChildERC20` is deployed on both the rootchain and the Supernet, acting as a minimal proxy for bridging ERC20 tokens between the two networks. In contrast, `NativeERC20` is deployed only on the Supernet and serves as the native token on the network.

Reviewer: ```suggestion
The main difference between `ChildERC20` and `NativeERC20` is their deployment and management. `ChildERC20` is deployed on both the rootchain and the Supernet, acting as a minimal proxy for bridging ERC20 tokens between the two networks. In contrast, `NativeERC20` is deployed only on the Supernet and serves as the native (gas) token on the network.
```
